enterprise/models.py

```diff
@@ -4259,35 +4264,107 @@ class Meta:
+        customer_id = str(self.enterprise_customer.pk).replace("-", "")
+        ecus = EnterpriseCustomerUser.objects.raw(sql_string, (customer_id, var_q))
+        return [ecu.id for ecu in ecus]
+
+    def _get_implicit_group_members(self, user_query=None):
+        """
+        Fetches all implicit members of a group, indicated by a (pending) enterprise customer user records.
+        """
+        members = []
+        # Regardless of user_query, we will need all pecus related to the group's customer
+        pending_customer_users = PendingEnterpriseCustomerUser.objects.filter(
+            enterprise_customer=self.enterprise_customer,
+        )
+
+        if user_query:
+            # Get all ecus relevant to the user query
             customer_users = EnterpriseCustomerUser.objects.filter(
-                enterprise_customer=self.enterprise_customer,
-                active=True,
+                id__in=self._get_filtered_ecu_ids(user_query)
             )
-            pending_customer_users = PendingEnterpriseCustomerUser.objects.filter(
+            # pecu has user_email as a field, so we can filter directly
+            pending_customer_users = pending_customer_users.filter(user_email__icontains=user_query)
+        else:
+            # No filtering query so get all ecus related to the group's customer
+            customer_users = EnterpriseCustomerUser.objects.filter(
                 enterprise_customer=self.enterprise_customer,
+                active=True,
             )
-            for ent_user in customer_users:
-                members.append(EnterpriseGroupMembership(
-                    uuid=None,
-                    enterprise_customer_user=ent_user,
-                    group=self,
-                ))
-            for pending_user in pending_customer_users:
-                members.append(EnterpriseGroupMembership(
-                    uuid=None,
-                    pending_enterprise_customer_user=pending_user,
-                    group=self,
-                ))
-            return members
+        # Build an in memory array of all the implicit memberships
+        for ent_user in customer_users:
+            members.append(EnterpriseGroupMembership(
+                uuid=None,
+                enterprise_customer_user=ent_user,
+                group=self,
+            ))
+        for pending_user in pending_customer_users:
+            members.append(EnterpriseGroupMembership(
+                uuid=None,
+                pending_enterprise_customer_user=pending_user,
+                group=self,
+            ))
+        return members
+
+    def _get_explicit_group_members(self, user_query=None, fetch_removed=False):
+        """
+        Fetch explicitly defined members of a group, indicated by an existing membership record
+        """
+        members = self.members.filter(is_removed=fetch_removed).select_related(
+            'enterprise_customer_user', 'pending_enterprise_customer_user'
+        )
```

Comment:
Just calling this out: if `fetch_removed` is True, these members will be _only_ the members that have been removed from the group.